feat(esl-animate): add media query based condition to disable animation (#3907)
exadel-inc/esl: packages/esl/src/esl-animate/README.md

### Service static and instance API:

- `ESLAnimateService.observe(targets, config)` — subscribe target element(s) for viewport-triggered animation.
  - `targets` — an Element or an array of Elements to observe
  - `config` — optional `ESLAnimateConfig` describing animation behavior for the targets
- `ESLAnimateService.unobserve(targets)` — unsubscribe target element(s) and remove internal tracking
- `ESLAnimateService.isObserved(el)` — returns `true` if the element is currently observed by the service

The static API is a convenience around a memoized shared service instance (so repeated calls operate on the same
service). You may also instantiate `ESLAnimateService` directly if you need a separate instance.

### Configuration API: `ESLAnimateConfig`
- `cls` (default: `in`) — CSS class(es) to add when an element becomes visible. Supports ESL extended class syntax
  (see [CSSClassUtils](../esl-utils/dom/class.ts)).

> - `group` (boolean) — enable grouped animation: items from the visibility queue will start with `groupDelay` between

Reviewer: ```suggestion
- `group` (boolean) — enable grouped animation: items from the visibility queue are animated sequentially with `groupDelay` between
```

  them.
- `groupDelay` (default: `100`) — delay in milliseconds between items when `group` is enabled.
- `repeat` (boolean) — if `true`, the item will be re-animated each time it exits and re-enters the viewport.
- `force` (boolean) — when `true`, the service will remove the animation class (if present) on subscribe so the

>   element can be animated again immediately after observation is attached.

Reviewer: ```suggestion
- `force` (boolean) — when `true`, removes the animation class (if present) when observation starts, allowing the element to be animated again immediately.
```

- `ratio` (default: `0.4`) — intersection ratio used to consider the element visible. Typical supported values are
  0.2, 0.4, 0.6, 0.8 (the service shares a single IntersectionObserver instance configured with thresholds

>   `[0.001, 0.2, 0.4, 0.6, 0.8, 1]`). Items are considered invisible when intersection is at or below ~1%.

Reviewer: ```suggestion
- `ratio` (default: `0.4`) — intersection ratio used to consider the element visible. Recommended values are `0.2`, `0.4`, `0.6`, and `0.8`. The service shares a single `IntersectionObserver` instance configured with thresholds `[0.001, 0.2, 0.4, 0.6, 0.8, 1]`. Elements are considered invisible when the intersection ratio is at or below ~1%.
```

> - `disableOn` — media-query condition (used by `ESLMediaQuery`) that disables animation when matched (default: `not all`).

Reviewer: ```suggestion
- `disableOn` — media-query condition (evaluated by `ESLMediaQuery`) that disables animations when matched (default: `not all`).
```

```js
  ESLAnimate.register();
```
### `ESLAnimate` Attributes | Properties:
- `target` — target element(s) to animate, defined by [ESLTraversingQuery](../esl-traversing-query/README.md). When

>   empty the `<esl-animate>` element itself is used (acts as a wrapper).

Reviewer: ```suggestion
  empty the `<esl-animate>` element itself is used (acting as a wrapper).
```

- `cls` (default: `in`) — CSS class(es) to add on visibility (supports ESL `CSSClassUtils`).
- `group` (boolean) — enable grouped animation for target items.
- `groupDelay` (default: `100`) — grouped animation delay in milliseconds.
- `repeat` (boolean) — re-animate items on each enter/exit cycle.
- `ratio` — intersection ratio to consider element visible (recommended: 0.2, 0.4, 0.6, 0.8; default: 0.4).
- `disableOn` — media-query condition that disables animation when matched (default: `not all`).
- `disableCls` — CSS class applied when animation is disabled (default: `esl-animate-inactive`).

By default, attributes `group`, `repeat`, `target`, `disableOn` and `disableCls` are observed and will trigger
reinitialization when changed. Call the instance method `reanimate()` to reinitialize observation manually; the element

> will request a forced re-animation when it reattaches targets.

Reviewer: ```suggestion
reinitialization when changed. Call the `reanimate()` instance method to manually reinitialize observation. During reattachment, targets are observed with forced re-animation enabled.
```
